source/slang/glsl.meta.slang

```diff
@@ -322,14 +322,18 @@ public vector<T,N> atan(vector<T,N> y, vector<T,N> x)
 }
 
 __generic<T : __BuiltinFloatingPointType>
-__target_intrinsic(cuda, "$P_asinh($0)")
-__target_intrinsic(cpp, "$P_asinh($0)")
 [__readNone]
 [ForceInline]
 [require(cpp_cuda_glsl_hlsl_spirv, GLSL_130)]
 public T asinh(T x)
 {
-    return log(x + sqrt(x * x + T(1)));
+    __target_switch
+    {
+    case cpp: __intrinsic_asm "$P_asinh($0)";
+    case cuda: __intrinsic_asm "$P_asinh($0)";
+    default:
+        return log(x + sqrt(x * x + T(1)));
```

Comment:
This is not a problem with the PR, although I did notice we implement many GLSL functions without a GLSL/SPIRV intrinsic path when such options exist (asinh/pack functions are an example)

An issue was made to address this (for SPIR-V). A comment is being added to link the PR to the issue for future reference: #4051

Reply:
I noticed it too.
Thank you for filing the issue.
I assigned it to myself.